pageserver/src/tenant.rs

```diff
@@ -487,6 +487,33 @@ enum CreateTimelineCause {
     Delete,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub(crate) enum GcError {
+    // The tenant is shutting down
+    #[error("tenant shutting down")]
+    TenantCancelled,
+
+    // The tenant is shutting down
+    #[error("timeline shutting down")]
+    TimelineCancelled,
+
+    // The tenant is in a state inelegible to run GC
+    #[error("not active")]
+    NotActive,
+
+    // A requested GC cutoff LSN was invalid, for example it tried to move backwards
+    #[error("not active")]
+    BadLsn { why: String },
+
+    // A remote storage error while scheduling updates after compaction
+    #[error(transparent)]
+    RemoteUpload(anyhow::Error),
+
+    // If GC was invoked for a particular timeline, this error means it didn't exist
+    #[error("timeline not found")]
+    NotFound,
```

Comment:
```suggestion
    TimelineNotFound,
```
if the *tenant* was not found we don't yield this error, right? because then from how I understand it, already the http handler errors. the rustdoc and error message describe this well but the name is a bit generic and easy to misunderstand.

Reply:
Right, when we call into GC it's always implicitly true that the tenant exists.  Changed in 950e5d235